tests/set_namespace_go.sh

```diff
@@ -41,17 +41,22 @@ metadata:
 data:
   "namespace": "example-ns"
 EOF
-kpt fn run .
-assert_contains_string example-configs/gatekeeper.yaml "namespace: example-ns"
-
-# Use the linux executable to test using kpt exec runtime on linux
-testcase "kpt_set_namespace_go_exec_imperative_linux"
-kpt pkg get "$SDK_REPO"/example-configs example-configs
-kpt pkg get "$CATALOG_REPO"/functions/go ./
-kpt fn run example-configs --enable-exec --exec-path "$(pwd)"/go/set-namespace/set-namespace-linux -- namespace=example-ns
-assert_contains_string example-configs/gatekeeper.yaml "namespace: example-ns"
+kpt fn source example-configs |
+  kpt fn run --fn-path fc.yaml >out.yaml
+assert_contains_string out.yaml "namespace: example-ns"
 
 testcase "kpt_set_namespace_go_declarative_example"
 kpt pkg get "$CATALOG_REPO"/examples/set-namespace .
 kpt fn run set-namespace/configs --fn-path set-namespace/functions
 assert_contains_string set-namespace/configs/example-config.yaml "namespace: example-ns"
+
+testcase "kpt_set_namespace_go_docker_imperative_idempotency"
+kpt pkg get "$SDK_REPO"/example-configs example-configs
```

Comment:
@mengqiy We should have all the example packages in the catalog repo. This is currently using examples in SDK repo which tightly couples the SDK and functions in the catalog.

Reply:
The first 2 tests for set-namespace are using sdk repo.

Reply:
They shouldn't. We should have all the example packages checked in the catalog repo. Otherwise, a PR in SDK repo can break tests here.

Reply:
https://github.com/GoogleContainerTools/kpt/issues/1248